packages/teleport/src/Player/Player.tsx

```diff
@@ -31,26 +31,47 @@ import { RecordingType } from 'teleport/services/recordings';
 
 export default function Player() {
   const { sid, clusterId } = useParams<UrlPlayerParams>();
+  const loc = useLocation();
 
   const recordingType = getUrlParameter(
     'recordingType',
-    useLocation().search
+    loc.search
   ) as RecordingType;
+  const durationMs = Number(getUrlParameter('durationMs', loc.search));
+
+  const validRecordingType =
+    recordingType === 'ssh' || recordingType === 'desktop';
+  // True if (recordingType !== 'desktop') because
+  // durationMs is only needed for desktop recordings.
+  const validDurationMs =
+    recordingType !== 'desktop' || Number.isInteger(durationMs);
```

Comment:
based on the comment `durationMs is only needed for desktop recordings`, shouldn't the condition then be:
```suggestion
    recordingType == 'desktop' && Number.isInteger(durationMs);
```
then we don't need the comment b/c the condition itself is self documenting?

i also think `validDurationMs` is misleading (or error prone depending on how you look at it) b/c it returns true for `recordingType !== 'desktop'` but doesn't check if its a valid integer.

Reply:
We want it to short circuit if `recordingType !== 'desktop'`. In that case, we're not using `durationMs` and so we don't care whether `Number.isInteger(durationMs) === true`.

Reply:
Still terrible design there, see latest commit.